data-transfer/src/main/java/it/eng/datatransfer/service/DataTransferService.java

```diff
@@ -158,8 +159,16 @@ public TransferProcess startDataTransfer(TransferStartMessage transferStartMessa
 		log.debug("Starting data transfer for consumerPid {} and providerPid {}", consumerPidFinal, providerPidFinal);
 
 		TransferProcess transferProcessRequested = findTransferProcess(consumerPidFinal, providerPidFinal);
+		
+		if (IConstants.ROLE_PROVIDER.equals(transferProcessRequested.getRole()) && TransferState.REQUESTED.equals(transferProcessRequested.getState())) {
```

Comment:
Do we have test cases for this logic?

Reply:
Yep, they were already there. I had to update them to work with the new changes